### PR TITLE
[doc] plugin implementation - bad direcoty for Entities

### PR DESCRIPTION
**Pull request by oallain**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12               |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| Related tickets |                      |
| License         | MIT                                                          |

In this page we mix `src/Model` and `src/Entity` IMO, we need clarify for beginners.

### CODE REVIEW

**Review by Prometee — DISMISSED** [review]

File: docs/book/plugins/guide/implementation.rst [hunk 1]
```diff
@@ -24,11 +24,11 @@ the same entity, it's recommended to provide a trait with new properties and met
 
     <?php
 
-    // src/Model/ProductVariantTrait.php
+    // src/Entity/ProductVariantTrait.php
 
     declare(strict_types=1);
 
-    namespace IronMan\SyliusProductOnDemandPlugin\Model;
+    namespace IronMan\SyliusProductOnDemandPlugin\Entity;
```
Comment (diimpp) [inline docs/book/plugins/guide/implementation.rst:32]: Trait provides annotation usage, but doesn't add `use`.

```patch
+++ use Doctrine\ORM\Mapping as ORM;
```

**Review by diimpp — COMMENTED** [review]
I do agree, that Model/Entity is confusing for very first timers.

But `Model` is a [default namespace](https://github.com/Sylius/SyliusResourceBundle/blob/5c0f0cc069b911af91b53aa3f392569060c41f6f/src/Bundle/AbstractResourceBundle.php#L91-L94) for entities, if doc-users won't override this method, then code snippet with `\Entity` will be an error.

```php
    protected function getModelNamespace(): ?string
    {
        return (new \ReflectionClass($this))->getNamespaceName() . '\\Model';
    }
```

If this change should be applied, then there should be mention of overriding `getModelNamespace` first. (Maybe it exists somewhere, I didn't check, if yes, then let's hyperlink it somehow to this doc)

**Review by zairigimad — APPROVED** [review]

**Comment by oallain** [comment]
Hello @diimpp ,

On Sylius, data is represented by Models because its storage is defined separately (in xml).

But in Sylius-Standard, we work [with Entities](https://github.com/Sylius/Sylius-Standard/tree/1.12/src/Entity), so I think we need to use Entities here to simplify things.

**Comment by diimpp** [comment]
@oallain Hi,

> But in Sylius-Standard, we work [with Entities](https://github.com/Sylius/Sylius-Standard/tree/1.12/src/Entity), so I think we need to use Entities here to simplify things.

As a plugin doc, it needs to be compared with [PluginTemplate](https://github.com/Sylius/PluginTemplate/blob/main/src/Plugin.php) or [PluginSkeleton](https://github.com/Sylius/PluginSkeleton/blob/1.12/src/AcmeSyliusExamplePlugin.php) where both don't re-configure `Model` to `Entity` path out of the box.

I agree with the change, but there should be **Tip** doc alert or some other way for readers to override `getModelNamespace` and switch to `Entity` path first.

If we make this change without extra notice about `getModelNamespace`, then example code snippet will not work out of the box -- my only concern is with this.

**Comment by jakubtobiasz** [comment]
I agree with @diimpp we can keep `Model` for now; especially it's used in the plugins' context. We'll rethink it during creating new docs. Thanks for your feedback!

**Pull request closed without merging**